app/controllers/usermatching/HomeController.scala

```diff
@@ -128,16 +128,18 @@ class HomeController @Inject()(val auditingService: AuditingService,
     withReference(utr) {
       reference =>
         subscriptionDetailsService.saveSubscriptionId(reference, mtditId) map {
-          _ =>
+          case Right(_) =>
             Redirect(controllers.individual.subscription.routes.ClaimSubscriptionController.claim)
               .withJourneyState(SignUp)
               .addingToSession(NINO -> nino)
               .addingToSession(UTR -> utr)
+          case Left(_) =>
+            throw new InternalServerException("[HomeController][claimSubscription] - Could not save subscription id")
```

Comment:
Is it not worth logging whatever is in the Left() ?

```suggestion
          case Left(error) =>
            throw new InternalServerException(s"[HomeController][claimSubscription] - Could not save subscription id: $error")
```